Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm

```diff
@@ -1356,9 +1359,19 @@ static UIDeviceOrientation toUIDeviceOrientation(DeviceOrientation orientation)
 void UIScriptControllerIOS::toggleCapsLock(JSValueRef callback)
 {
     m_capsLockOn = !m_capsLockOn;
-    auto *keyboardEvent = createUIPhysicalKeyboardEvent(@"capsLock", [NSString string], m_capsLockOn ? UIKeyModifierAlphaShift : 0,
-        kUIKeyboardInputModifierFlagsChanged, m_capsLockOn);
-    [[UIApplication sharedApplication] handleKeyUIEvent:keyboardEvent];
+    auto [keyboardEventDown, keyboardEventUp] = [capsLockOn = m_capsLockOn] {
+        auto *keyboardEventDown = createUIPhysicalKeyboardEvent(@"capsLock", [NSString string], capsLockOn ? UIKeyModifierAlphaShift : 0, kUIKeyboardInputModifierFlagsChanged, IsKeyDown::Yes);
+        auto *keyboardEventUp = createUIPhysicalKeyboardEvent(@"capsLock", [NSString string], capsLockOn ? UIKeyModifierAlphaShift : 0, kUIKeyboardInputModifierFlagsChanged, IsKeyDown::No);
+        return std::make_pair(keyboardEventDown, keyboardEventUp);
+    }();
+    auto *press = [&keyboardEvent = keyboardEventDown] {
```

Comment:
This whole dance because it's illegal to (explicitly or implicitly) capture a structured binding in C++
https://burnicki.pl/en/2021/04/19/capture-structured-bindings.html

Reply:
Why do we need lambdas here? (as opposed to just using local variables)